src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupped: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    if (!groupped[String(item[key])]) {
+      groupped[String(item[key])] = [];
+    }
+
+    groupped[String(item[key])].push(item);
```

Comment:
You repeat `String(item[key])` several time, so it is reasonable to create a var for it